pkg/prometheus/agent/statefulset.go

```diff
@@ -45,6 +45,7 @@ func makeStatefulSet(
 	inputHash string,
 	shard int32,
 	tlsSecrets *operator.ShardedSecret,
+	controllerID string,
```

Comment:
I don't think that we need to propagate the controller ID to the statefulset and pods. Let's keep it simple for now.

Reply:
done